refactor: platform-neutral API cleanup - rename Android-specific symbols and remove vendor path hardcoding

quic-vkatoch opened the pull request:
Remove Android-specific naming, dead code, and hardcoded paths from the fastrpc library to make it platform-neutral.
  - Rename listener_android.h/c to listener.h/c and update all symbol references throughout the codebase.
  - Replace fastrpc_android.h/c with fastrpc_platform.h/c exposing a neutral two-function API; move Android internals to file-static scope and add an LE stub in the same file.
  - Fix off-by-one OOB bounds check in fastrpc_get_property_int/string (> to >=) for the FASTRPC_BUILD_TYPE property key.
  - Remove hardcoded /vendor/dsp/ fallback paths from open_shell(); these are already covered by DEFAULT_DSP_SEARCH_PATHS at build time.
  - Fix header guard FASTRPC_ANDROID_USER_H to FASTRPC_APPS_USER_H.
  - Remove dead #if 0 blocks, stale comments, and Android-specific arrays that are no longer reachable after the platform layer refactor.

quic-vkatoch committed:
Remove hardcoded vendor DSP path fallbacks

Remove VENDOR_DSP_LOCATION and VENDOR_DOM_LOCATION macros from
apps_std_internal.h and fastrpc_apps_user.c.

Simplify open_shell() by removing fallback logic that attempted
to open shell files from /vendor/dsp/ and /vendor/dsp/xdsp/.
Shell files are now opened only from DSP_LIBS_LOCATION paths.

Update comments in apps_std_imp.c to reflect generic fallback
behavior without vendor path references.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

quic-ekangupt commented on src/fastrpc_apps_user.c:
```diff
@@ -3548,12 +3483,6 @@ static int open_shell(int domain_id, apps_std_FILE *fh, int unsigned_shell) {
       nErr = AEE_SUCCESS;
       *fh = -1;
-    } else {
-      FARF(ERROR,
```
should not completely remove the error log

lumag replied:
So, the error log needs to be fixed rather than being completely removed.

quic-vkatoch replied:
Okay, I'll only be logging the paths that have been actually checked.

quic-ekangupt commented on src/adspmsgd.c:
```diff
@@ -1,176 +1,176 @@
+#endif
+
+#include <string.h>
+#include <errno.h>
```
can you highlight what is changed in this file?

lumag commented on src/fastrpc_apps_user.c:
```diff
@@ -241,16 +230,6 @@ const char *ENV_DEBUG_VAR_NAME[] = {"FASTRPC_PROCESS_ATTRS",
-                                         "vendor.fastrpc.debug.pddump",
-                                         "persist.vendor.fastrpc.process.attrs",
-                                         "ro.build.type"};
 const char *ANDROID_DEBUG_VAR_NAME[] = {"fastrpc.process.attrs",
```
This is also Android-specific var and was missed

lumag commented on src/fastrpc_apps_user.c:
```diff
@@ -241,26 +230,6 @@ const char *ENV_DEBUG_VAR_NAME[] = {"FASTRPC_PROCESS_ATTRS",
                                     "FASTRPC_DEBUG_PDDUMP",
                                     "FASTRPC_PROCESS_ATTRS_PERSISTENT",
                                     "ro.debuggable"};
```
`ro.debuggable` is also an Android definition.

quic-vkatoch replied:
I'll check and remove it.

quic-ekangupt commented:
Please check if you can break the change into multiple commits instead of a single commit.

quic-vkatoch committed:
Remove Android reference from fastrpc_apps_user.h header guards

Change header guard macros from FASTRPC_ANDROID_USER_H to
FASTRPC_APPS_USER_H to remove Android-specific naming and
make the header more generic for cross-platform usage.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

quic-vkatoch committed:
Rename listener_android to listener and update all references

Rename listener_android.h to listener.h and listener_android.c
to listener.c to remove Android-specific naming. Update all
function names from listener_android_* to listener_* pattern.
Update header guards, includes, Makefile entries, platform
library dependencies, and all function calls throughout the
codebase to maintain compilation compatibility.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

quic-vkatoch committed:
Clean up unused code, TODO comments, and Android references

Remove unused convert_level_to_android_priority() function,
TODO comments referencing Android-specific code, and Android
include statements. Update AEE_EINVHANDLE comment to remove
vendor-specific reference and clean up conditional compilation
blocks for Android platforms.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

quic-vkatoch committed:
Move Android property handling to fastrpc_android

Create fastrpc_android.h and fastrpc_android.c with Android-specific
property handling. Move ANDROIDP_DEBUG_VAR_NAME array and implement
platform_get_property_int() and platform_get_property_string().

Remove ANDROID_DEBUG_VAR_NAME array and SYSTEM_RPC_LIBRARY conditional
compilation from fastrpc_apps_user.c. Simplify fastrpc_get_property_int()
and fastrpc_get_property_string() to call platform functions.

Update Makefiles to include fastrpc_android.h and conditionally compile
fastrpc_android.c for Android builds.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

lumag commented on src/fastrpc_log.c:
```diff
@@ -288,17 +287,7 @@ void fastrpc_log_init() {
-#endif
-  }
-  if (persist_buf.buf == NULL && debug_build_type) {
+  if (persist_buf.buf == NULL) {
```
These are separate changes. Why are you squashing them into a single commit?

quic-vkatoch replied:
I have separated the changes into multiple commits and it's no longer a single commit. I'm checking if it could be separated any further.

quic-ekangupt commented:
@quic-vkatoch, can you please rebase this PR?

ekanshibu commented on src/fastrpc_log.c:
```diff
@@ -1,15 +1,13 @@
 // Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
-#define _GNU_SOURCE
```
why removing this?

ekanshibu commented on inc/AEEStdErr.h:
```diff
@@ -84,7 +84,7 @@
 #define AEE_EINVALIDITEM              (AEE_EOFFSET + 0x02A)  ///< Current item is invalid, it can be a switch case or a pointer to memory
 #define AEE_ENOTALLOWED               (AEE_EOFFSET + 0x02B)  ///< Not allowed to perform the
```
fix all commit messages

quic-vkatoch commented on inc/apps_std_internal.h:
I have to check where the shell should be placed if /vendor/dsp support is removed, or if we actually need to remove it in the first place.

Regarding testing: earlier, the daemons were already running (opened the shell) and the tests passed. After killing and updating with new daemons, the tests are now failing. I will check this further and update.

lumag replied:
> If we are removing the support for /vendor/dsp, where is shell expected to be pushed for android targets ? if it is /usr/lib/dsp/ like other LE targets, is /usr partition available on android targets ?

All DSP binaries are located in the subdirs of `/usr/share/qcom`, one subdir per device.

quic-vkatoch commented:
@lumag, could you please review this PR?

quic-vkatoch commented on src/fastrpc_android.c:
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "fastrpc_android.h"
```
Done. Moved to a later commit.

quic-bkumar commented on inc/fastrpc_android.h:
Are we okay to name the file with android in it?

quic-vkatoch replied:
Updated, to make it platform-neutral

quic-vkatoch committed:
refactor: rename listener_android to listener

Rename listener_android.h/c to listener.h/c and update all symbol
references (listener_android_* to listener_*), PL_DEFINE/PL_DEP/PL_ENTRY
entries, and Makefiles.

Signed-off-by: Vinayak Katoch <vinayak.katoch@oss.qualcomm.com>

quic-vkatoch committed:
refactor: replace fastrpc_android with platform-neutral fastrpc_platform

Replace fastrpc_android.h/c with fastrpc_platform.h/c exposing two
neutral functions: fastrpc_platform_get_property_int/string. Android
internals (property name table, property_get stubs) moved to file-static
scope; LE stub added in the same file under #else. Update
fastrpc_apps_user.c to include the new header unconditionally, remove
Android-specific arrays and stubs, fix an off-by-one OOB bounds check
(> to >=) in the property key range check, and update Makefiles.

Signed-off-by: Vinayak Katoch <vinayak.katoch@oss.qualcomm.com>

quic-vkatoch committed:
refactor: remove hardcoded vendor DSP path fallbacks

Remove VENDOR_DSP_LOCATION and VENDOR_DOM_LOCATION defines and the
vendor path fallback logic in open_shell(). Inline the DSP_LIBS_LOCATION
search directly into open_shell() removing the try_open_shell_file()
helper which existed solely to abstract the vendor fallbacks. The
base-name retry (fastrpc_shell without domain suffix) is preserved.
Also update two vendor-specific comments in apps_std_imp.c and remove
a dead #if 0 block.

Signed-off-by: Vinayak Katoch <vinayak.katoch@oss.qualcomm.com>

quic-vkatoch committed:
cleanup: remove dead code and fix Android-specific leftovers

Remove stale AEE_EINVHANDLE comment, fix header guard
FASTRPC_ANDROID_USER_H to FASTRPC_APPS_USER_H, remove dead Android TODO
include block in fastrpc_trace.h, remove #if 0 log priority function in
adspmsgd_printf.c, fix "buuid" typo in fastrpc_log.c, and collapse
3-way PROPERTY_VALUE_MAX #ifdef to a single #define in fastrpc_mem.c.

Signed-off-by: Vinayak Katoch <vinayak.katoch@oss.qualcomm.com>
